lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -294,8 +308,29 @@ def write(chunk)
             end
           end
         end
+        if @running_cloudfunctions
+          # If the current group of entries is coming from a Cloud Functions
+          # function, the function name can be extracted from the tag.
+          match_data = @cloudfunctions_tag_regexp.match(tag)
+          if match_data
+            # Service name is set to Cloud Functions only for logs actually
+            # coming from a function.
+            @service_name = CLOUDFUNCTIONS_SERVICE
+            labels = write_log_entries_request['commonLabels']
+            labels["#{CLOUDFUNCTIONS_SERVICE}/function_name"] =
+              match_data['function_name']
+          else
+            # Other logs are considered as coming from the Container Engine
+            # service.
+            @service_name = CONTAINER_SERVICE
+          end
+        end
         arr.each do |time, record|
           next unless record.is_a?(Hash)
+          if @service_name == CLOUDFUNCTIONS_SERVICE && record.key?('log')
```

Comment:
The format of 'log' will be rather static although it is possible that it will evolve over time. I could potentially include a stringified JSON as the 'log' string value and parse it in the plugin, if this is what you mean? I would still need to check if this is what we want, though.

Emitting a true JSON object as the 'log' value may not be possible as the overall composition of the emitted records is rather not controllable.